hathor/websocket/factory.py

```diff
@@ -314,14 +332,42 @@ def _remove_connection_from_address_dict(self, connection: HathorAdminWebsocketP
         if len(self.address_connections[address]) == 0:
             del self.address_connections[address]
 
+    def _handle_subscribe(self, connection: HathorAdminWebsocketProtocol, message: Dict[Any, Any]) -> None:
+        """ Handler for subscription to a ws message."""
+        ws_message: Optional[str] = message.get('message')
+        if ws_message is None:
+            # No message key
+            return
+        if ws_message in WS_MESSAGES:
+            if connection in self.connections:
+                # Remove from the all connections set
+                # Only in the first message subscription will still be in the set
+                self.connections.remove(connection)
+            # Add the connection to the set
+            self.subscribed_connections[ws_message].add(connection)
+        # Reply back as subscribed success
+        payload = json.dumps({'type': 'subscribed', 'message': ws_message, 'success': True}).encode('utf-8')
+        connection.sendMessage(payload, False)
+
     def connection_closed(self, connection: HathorAdminWebsocketProtocol) -> None:
         """ Called when a ws connection is closed
             We should remove it from self.connections and from self.address_connections set for each address
         """
-        self.connections.remove(connection)
+        if connection in self.connections:
+            # With connections subscribing to an specific channel, we need to do this check
+            # In the future all connections should subscribe to channels
+            self.connections.remove(connection)
+
         for address in connection.subscribed_to:
             self._remove_connection_from_address_dict(connection, address)
 
+        # Removing from subsbribed channels set
+        # Should we have an easier way to check in each channels the connections is subscribed to?
+        # For now we have only one channel, so this shouldn't be a problem
```

Comment:
Comment is out of date (channels).

Given that we don't have many message types and use a set for storing the connections, shouldn't be a very expensive operation.

Reply:
Updated the comments there and added your comment also, I agree it's not expensive to remove from all sets.